### PR TITLE
refactor builtin string function ord

### PR DESCRIPTION
**Pull request by wusphinx**
Refactor builtin string function `ORD`

I implemented the function `ORD` and added some test cases in executor

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1639,49 +1639,40 @@ type ordFunctionClass struct {
 }
 
 func (c *ordFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinOrdSig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = 10
```
Comment (zz-jason) [inline expression/builtin_string.go]: why set `bf.tp.Flen` to `10` ?

File: executor/executor_test.go [hunk 2]
```diff
@@ -841,6 +841,15 @@ func (s *testSuite) TestStringBuiltin(c *C) {
 	result.Check(testkit.Rows("-1 1 -1 <nil> <nil>"))
 	result = tk.MustQuery(`select strcmp("", "123"), strcmp("123", ""), strcmp("", ""), strcmp("", null), strcmp(null, "")`)
 	result.Check(testkit.Rows("-1 1 0 <nil> <nil>"))
+
+	// for ord
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a char(10), b int, c double, d datetime, e time, f bit(4))")
```
Comment (zz-jason) [inline executor/executor_test.go]: add binary-series types, like `binary(20)` or `blob(10)`

File: expression/builtin_string.go (hunk 1 above)
Comment (wusphinx) [inline expression/builtin_string.go]: Because the max length of  multibyte character is 4 byte, which contains up to 10 numbers, so I set it to 10.

Comment (zz-jason) [inline expression/builtin_string.go]: i c

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

**Comment by shenli** [comment]
LGTM
@wusphinx Thanks!

**Pull request merged**